nilearn/maskers/nifti_labels_masker.py

```diff
@@ -188,8 +192,14 @@ def __init__(
         **kwargs,
     ):
         self.labels_img = labels_img
-        self.labels = labels
+
         self.background_label = background_label
+        self._original_region_ids = self._get_labels_values(self.labels_img)
```

Comment:
introducing a new private attribute to keep track of the labels of the image:
- this allows to check that the labels and number of regions match at instantiation
- should later allow to know which regions were dropped during resampling